store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -192,4 +192,48 @@ public void modifyServerProperties(VolumeInfo volInfo) throws ServiceException,
             throw e;
         }
     }
+
+    /**
+     * Validate global bucket ID by reading LDAP atrribute
+     * @param globalS3BucketId
+     * @returns true if "global bucket ID" is valid
+     * @throws ServiceException, JSONException
+     */
+    public Boolean validateGlobalBucketID(String globalS3BucketId) throws ServiceException {
+        try {
+            // step 1: Fetch globalS3Configs and globalS3ConfigList
+            String globalExternalStoreConfig = provisioning.getConfig().getGlobalExternalStoreConfig();
+            JSONObject globalS3Configs = new JSONObject(globalExternalStoreConfig);
+            JSONArray globalS3ConfigList = globalS3Configs.getJSONArray("global/s3BucketConfigurations");
+
+            // step 2: Find "globalBucketUUID" in current JSON array
+            for (int i = 0; i < globalS3ConfigList.length(); i++) {
+                // step 3: Mark validation as true if "globalBucketUUID" found
+                if (globalS3BucketId.equalsIgnoreCase(globalS3ConfigList.getJSONObject(i).getString("globalBucketUUID"))) {
+                    return true;
+                }
+            }
+        } catch (JSONException e) {
+            throw ServiceException.FAILURE("Error while validating GlobalBucketID", null);
+        }
+        return false;
+    }
+
+    /**
+     * Validate if external volume entry is present in JSON or not
+     * @param volumeId
+     * @returns true if external volume entry is present
+     * @throws ServiceException, JSONException
+     */
+    public Boolean isVolumePresentInJson(int volumeId) throws ServiceException, JSONException {
+      String serverS3ConfigJSON = provisioning.getLocalServer().getServerExternalStoreConfig();
+      JSONObject serverS3ConfigJSONObject = new JSONObject(serverS3ConfigJSON);
```

Comment:
can we rename serverS3ConfigJSONObject &  serverS3ConfigJSONArray these as well

Reply:
Fixed